server/router.js

```diff
@@ -14,16 +14,16 @@ export default class Router {
     this.routes.set(method, routes)
   }
 
-  match (req, res) {
+  match (req, res, url = parse(req.url)) {
```

Comment:
We don't need to parse the url here.
`.match()` is always called with a parsedUrl.